src/solvers/thermal_operators.cpp

```diff
@@ -10,46 +10,44 @@
 
 namespace serac {
 
-DynamicConductionOperator::DynamicConductionOperator(std::shared_ptr<mfem::ParFiniteElementSpace> fespace,
-                                                     const serac::LinearSolverParameters&         params,
-                                                     std::vector<serac::BoundaryCondition>&       ess_bdr)
-    : mfem::TimeDependentOperator(fespace->GetTrueVSize(), 0.0),
-      fespace_(fespace),
+DynamicConductionOperator::DynamicConductionOperator(mfem::ParFiniteElementSpace&           fespace,
+                                                     const serac::LinearSolverParameters&   params,
+                                                     std::vector<serac::BoundaryCondition>& ess_bdr)
+    : mfem::TimeDependentOperator(fespace.GetTrueVSize(), 0.0),
       ess_bdr_(ess_bdr),
-      z_(fespace->GetTrueVSize()),
-      y_(fespace->GetTrueVSize()),
-      x_(fespace->GetTrueVSize()),
+      z_(fespace.GetTrueVSize()),
+      y_(fespace.GetTrueVSize()),
+      x_(fespace.GetTrueVSize()),
       old_dt_(-1.0)
 {
   // Set the mass solver options (CG and Jacobi for now)
-  M_solver_ = AlgebraicSolver(fespace_->GetComm(), params);
+  M_solver_ = AlgebraicSolver(fespace.GetComm(), params);
```

Comment:
I guess I didn't get my review of the other PR in time, but `AlgebraicSolver` seems like a bad name to me. Algebraic equation has a specific meaning mathematically that I don't think applies here. For example, 
`exp(x) + x == 0` is not an algebraic equation, but is still something that we would expect to be able to solve with that class, I imagine.

Reply:
Yeah, the original idea was `SystemSolver`, but in my mind that comes with a notion of a full physics solver. The idea is this object is reduced to a simple mathematical form. `AlgebraicSolver` was the best I could come up with unfortunately but I definitely agree with your point. Any other naming thoughts?

Reply:
`ExpressionSolver`? `EquationSolver`?

Reply:
I'm not sure what this object really does.

Reply:
Maybe EquationSolver, NonlinearSystemSolver, or RootFinder?

Reply:
It can be either linear or nonlinear (`LinearAlgebraSolver` was considered earlier), but maybe `EquationSolver` or `EquationSystemSolver` could work?

Reply:
I like either `EquationSolver` or `EquationSystemSolver` with a slight preference for the former for brevity.